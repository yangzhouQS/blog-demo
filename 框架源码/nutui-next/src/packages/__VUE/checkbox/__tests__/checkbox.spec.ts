import { mount } from '@vue/test-utils';
import Checkbox from '../index.vue';

test('prop label', () => {
  const wrapper = mount(Checkbox, {
    props: {
      label: '复选框checkbox'
    }
  });
  expect(wrapper.html()).toMatchSnapshot();
});
test('prop disabled', () => {
  const wrapper = mount(Checkbox, {
    props: {
      modelValue: true,
      disabled: true
    }
  });
  expect(wrapper.html()).toMatchSnapshot();
});
test('prop iconSize', () => {
  const wrapper = mount(Checkbox, {
    props: {
      modelValue: true,
      iconSize: '20'
    }
  });
  expect(wrapper.html()).toMatchSnapshot();
});
test('prop iconName iconActiveName', () => {
  const wrapper = mount(Checkbox, {
    props: {
      iconName: 'check-normal',
      iconActiveName: 'checked'
    }
  });
  expect(wrapper.html()).toMatchSnapshot();
});

test('should emit "update:modelValue" event when checkbox is clicked', async () => {
  const wrapper = mount(Checkbox);

  wrapper.trigger('click');
  expect(wrapper.emitted('update:modelValue')![0]).toEqual([true]);

  await wrapper.setProps({ modelValue: true });
  wrapper.trigger('click');
  expect(wrapper.emitted('update:modelValue')![1]).toEqual([false]);
});
