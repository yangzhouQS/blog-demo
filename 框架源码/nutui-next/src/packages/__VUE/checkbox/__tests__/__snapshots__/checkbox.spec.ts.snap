// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prop disabled 1`] = `
"<view class=\\"nut-checkbox \\"><i class=\\"nutui-iconfont nut-icon nut-icon-checked nut-checkbox__icon--disable\\" style=\\"font-size: 18px; width: 18px; height: 18px;\\" src=\\"\\"></i>
  <view class=\\"nut-checkbox__label nut-checkbox__label--disabled\\"></view>
</view>"
`;

exports[`prop iconName iconActiveName 1`] = `
"<view class=\\"nut-checkbox \\"><i class=\\"nutui-iconfont nut-icon nut-icon-check-normal nut-checkbox__icon--unchecked\\" style=\\"font-size: 18px; width: 18px; height: 18px;\\" src=\\"\\"></i>
  <view class=\\"nut-checkbox__label \\"></view>
</view>"
`;

exports[`prop iconSize 1`] = `
"<view class=\\"nut-checkbox \\"><i class=\\"nutui-iconfont nut-icon nut-icon-checked nut-checkbox__icon\\" style=\\"font-size: 20px; width: 20px; height: 20px;\\" src=\\"\\"></i>
  <view class=\\"nut-checkbox__label \\"></view>
</view>"
`;

exports[`prop label 1`] = `
"<view class=\\"nut-checkbox \\"><i class=\\"nutui-iconfont nut-icon nut-icon-check-normal nut-checkbox__icon--unchecked\\" style=\\"font-size: 18px; width: 18px; height: 18px;\\" src=\\"\\"></i>
  <view class=\\"nut-checkbox__label \\"></view>
</view>"
`;
